tests/testthat/test-expect_null_linter.R

```diff
@@ -0,0 +1,36 @@
+test_that("expect_null_linter skips allowed usages", {
+  # NB: this usage should be expect_false(), but this linter should
+  #   nevertheless apply (e.g. for maintainers not using ExpectNotLinter)
+  expect_lint("expect_true(!is.null(x))", NULL, expect_null_linter())
+  # NB: also applies to tinytest, but it's sufficient to test testthat
+  expect_lint(
+    "testthat::expect_true(!is.null(x))",
+    NULL,
+    expect_null_linter()
+  )
+
+  # length-0 could be NULL, but could be integer() or list(), so let it pass
+  expect_lint("expect_length(x, 0L)", NULL, expect_null_linter())
+})
+
+test_that("expect_null_linter blocks simple disallowed usages", {
+  expect_lint(
+    "expect_equal(x, NULL)",
```

Comment:
`expect_equal(NULL, x)` should also lint I think

Reply:
for us I _believe_ this is covered by the `yoda_test_linter` (if not, it should be), but yes, we should generally write linters that work independently of having others on at the same time. Will do.